cmd/kubelet/app/server.go

```diff
@@ -375,13 +371,10 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 			if err != nil {
 				glog.Warningf("Failed to create API Server client for heartbeat: %v", err)
 			}
-		} else {
-			switch {
-			case s.RequireKubeConfig:
-				return fmt.Errorf("invalid kubeconfig: %v", err)
-			case s.KubeConfig.Provided():
-				glog.Warningf("invalid kubeconfig: %v", err)
-			}
+		}
+
+		if err != nil && len(s.KubeConfig) > 0 {
```

Comment:
can drop this completely... in cases where we needed something from the kubeconfig, we already logged it above

Reply:
oops, misread nesting... keep the logging, and no need to check s.KubeConfig... if we're in this block, it was provided